core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchActivity.java

```diff
@@ -142,6 +149,10 @@ public boolean onCreateOptionsMenu(Menu menu) {
       activateAutoAdapter();
       autoAdapter.getFilter().filter(searchText.toLowerCase());
     }
+    if (getIntent().getStringExtra(EXTRA_COME_FROM).equals("TAB_SWITCHER")) {
```

Comment:
this comparison gets repeated , you should adhere to the DRY principle. You could create a method that does it but storing the result of this in a field variable is also acceptable